File: shifai-android/app/src/main/java/com/shifai/presentation/accessibility/AccessibilityLabels.kt
package com.shifai.presentation.accessibility

/**
 * Accessibility Labels — TalkBack content descriptions in French.
 * Centralized for consistent accessibility across all screens.
 * Mirrors iOS AccessibilityLabels.swift.
 */
object AccessibilityLabels {

    // ─── Dashboard ───

    object Dashboard {
        const val CYCLE_DAY = "Jour du cycle"
        const val PHASE_INDICATOR = "Phase du cycle actuelle"
        const val NEXT_PREDICTION = "Prochaine prédiction"
        const val MOOD_SCORE = "Score d'humeur"
        const val ENERGY_PROGRESS = "Niveau d'énergie"
    }

    // ─── Tracking ───

    object Tracking {
        const val FLOW_SLIDER = "Intensité du flux"
        const val FLOW_HINT = "Ajustez entre 0 et 4"
        const val MOOD_SLIDER = "Humeur"
        const val MOOD_HINT = "Ajustez entre 1 et 10"
        const val SLEEP_SLIDER = "Heures de sommeil"
        const val SLEEP_HINT = "Ajustez entre 0 et 24 heures"
        const val SAVE_BUTTON = "Enregistrer les données du jour"
        const val BODY_MAP = "Carte corporelle interactive"
        const val BODY_MAP_HINT = "Appuyez pour sélectionner une zone de symptôme"
    }

    // ─── Insights ───

    object Insights {
        const val INSIGHT_CARD = "Carte d'analyse"
        const val FILTER_MENU = "Filtrer les analyses"
        const val FEEDBACK_POSITIVE = "Marquer comme utile"
        const val FEEDBACK_NEGATIVE = "Marquer comme pas utile"
        const val UNREAD_BADGE = "Analyses non lues"
    }

    // ─── Settings ───

    object Settings {
        const val SYNC_TOGGLE = "Synchronisation automatique"
        const val BIOMETRIC_TOGGLE = "Verrouillage biométrique"
        const val EXPORT_BUTTON = "Exporter mes données"
        const val DELETE_ACCOUNT = "Supprimer mon compte"
        const val DELETE_HINT = "Action irréversible. Toutes vos données seront supprimées."
    }

    // ─── Common ───

    object Common {
        const val LOADING = "Chargement en cours"
        const val ERROR = "Une erreur est survenue"
        const val RETRY = "Réessayer"
        const val BACK = "Retour"
        const val CLOSE = "Fermer"
    }
}

File: shifai-ios/ShifAI/Presentation/Accessibility/AccessibilityLabels.swift
import SwiftUI

/// Accessibility Helpers — VoiceOver labels and hints in French.
/// Centralized to ensure consistent accessibility across all views.
enum AccessibilityLabels {

    // MARK: - Dashboard

    enum Dashboard {
        static let cycleDay = "Jour du cycle"
        static let phaseIndicator = "Phase du cycle actuelle"
        static let nextPrediction = "Prochaine prédiction"
        static let moodScore = "Score d'humeur"
        static let energyProgress = "Niveau d'énergie"
    }

    // MARK: - Tracking

    enum Tracking {
        static let flowSlider = "Intensité du flux"
        static let flowHint = "Ajustez entre 0 et 4"
        static let moodSlider = "Humeur"
        static let moodHint = "Ajustez entre 1 et 10"
        static let sleepSlider = "Heures de sommeil"
        static let sleepHint = "Ajustez entre 0 et 24 heures"
        static let saveButton = "Enregistrer les données du jour"
        static let bodyMap = "Carte corporelle interactive"
        static let bodyMapHint = "Appuyez pour sélectionner une zone de symptôme"
    }

    // MARK: - Insights

    enum Insights {
        static let insightCard = "Carte d'analyse"
        static let filterMenu = "Filtrer les analyses"
        static let feedbackPositive = "Marquer comme utile"
        static let feedbackNegative = "Marquer comme pas utile"
        static let unreadBadge = "Analyses non lues"
    }

    // MARK: - Settings

    enum Settings {
        static let syncToggle = "Synchronisation automatique"
        static let biometricToggle = "Verrouillage biométrique"
        static let exportButton = "Exporter mes données"
        static let deleteAccount = "Supprimer mon compte"
        static let deleteHint = "Action irréversible. Toutes vos données seront supprimées."
    }

    // MARK: - Common

    enum Common {
        static let loading = "Chargement en cours"
        static let error = "Une erreur est survenue"
        static let retry = "Réessayer"
        static let back = "Retour"
        static let close = "Fermer"
    }
}
